homeassistant/core.py

```diff
@@ -217,13 +217,14 @@ class HassJob(Generic[_P, _R_co]):
     we run the job.
     """
 
-    __slots__ = ("job_type", "target", "name")
+    __slots__ = ("job_type", "target", "name", "cancel_on_shutdown")
 
     def __init__(self, target: Callable[_P, _R_co], name: str | None = None) -> None:
         """Create a job object."""
         self.target = target
         self.name = name
         self.job_type = _get_hassjob_callable_job_type(target)
+        self.cancel_on_shutdown: bool | None = None
```

Comment:
Let's make this private and only accept it at construction time so we don't get have an action at a distance anti pattern when the flag gets flipped by something passing around the job